llama-index-core/llama_index/core/schema.py

```diff
@@ -357,6 +357,9 @@ def as_related_node_info(self) -> RelatedNodeInfo:
 
 class TextNode(BaseNode):
     text: str = Field(default="", description="Text content of the node.")
+    mimetype: str = Field(
+        default="text/plain", description="MIME type of the node content."
```

Comment:
do we want an enum for this, or is a simple string good enough?

Reply:
the reason for string is that mimetypes are open ended and also can have parameters
```
type/subtype;parameter=value
```

from [MDN](https://developer.mozilla.org/en-US/docs/Web/HTTP/Basics_of_HTTP/MIME_types)

Reply:
look at this mimetype: this is for [json tables](https://specs.frictionlessdata.io//tabular-data-resource/)


[application/vnd.dataresource+json](https://frictionlessdata.io/specs/tabular-data-resource/)

```js
{
  "profile": "tabular-data-resource",
  "name": "resource-name",
  "data": [
    {
      "id": 1,
      "first_name": "Louise"
    },
    {
      "id": 2,
      "first_name": "Julia"
    }
  ],
  "schema": {
    "fields": [
      {
        "name": "id",
        "type": "integer"
      },
      {
        "name": "first_name",
        "type": "string"
      }
    ],
    "primaryKey": "id"
  }
}
```

Reply:
lol gotcha -- point proven!